src/macrofier/engine.mjs

```diff
@@ -133,6 +134,24 @@ const getLinkForSchema = (schema, json, { name = '' } = {}) => {
   return '#'
 }
 
+const getComponentExternalSchema = (json) => {
```

Comment:
You _might_ be able to use `getExternaSchemaPaths` from `shared/json-schema.mjs` instead of making a new method. Not sure if there's a minor difference, but worth considering.

Reply:
Actually this was added to get actual dependency of all common schema cases. Some common schema have dependency with common schema itself, say Discovery, Intents have dependency with Entertainment Schema and Entertainment has dependency with types, but as I mentioned above 'x-schemas' in common schema is not pointing its exact dependency. so this method is added to find out exact dependency of all components inside common schema.

Reply:
got it, makes sense.